src/com/goide/runconfig/testing/coverage/GoCoverageProcessListener.java

```diff
@@ -0,0 +1,35 @@
+package com.goide.runconfig.testing.coverage;
+
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.execution.process.ProcessListener;
+import com.intellij.openapi.util.Key;
+
+import java.io.File;
+
+class GoCoverageProcessListener implements ProcessListener {
+  private String myCoverageFilePath;
+  private String myCoverageDir;
+
+  public GoCoverageProcessListener(String fromDir, String toFile) {
+    myCoverageFilePath = toFile;
+    myCoverageDir = fromDir;
+  }
+
+  @Override
+  public void startNotified(ProcessEvent event) {}
+
+  @Override
+  public void processTerminated(ProcessEvent event) {
+    if (event.getExitCode() != 0 || new File(myCoverageFilePath).exists()) {
+      return;
+    }
+
+    GoCoverageMerger.MergeCoverage(myCoverageDir, myCoverageFilePath);
```

Comment:
Please make it OOP. Merger that just contain some static classes is not Merger but some Utility class.